src/NServiceBus.Core/BusConfiguration.cs

```diff
@@ -267,11 +267,18 @@ internal InitializableEndpoint Build()
         {
             ForAllTypes<T>(types, t =>
             {
+                if (!HasDefaultConstructor(t))
+                {
+                    throw new Exception($"Unable to create the type '{t.Name}'. Types implementing '{typeof(T).Name}' must have a public parameterless (default) constructor.");
+                }
+
                 var instanceToInvoke = (T)Activator.CreateInstance(t);
                 action(instanceToInvoke);
             });
         }
 
+        static bool HasDefaultConstructor(Type type) => type.GetConstructor(Type.EmptyTypes) != null;
```

Comment:
I'm not 100% sure of the internals, but I think `Activator` does a better job of creating instances: it does additional checks and caches some results (according to the reference source). I could change that to the `ctor.Invoke()`, but I don't see much benefit in doing that.